lang/ruby/lib/avro/io.rb

```diff
@@ -318,7 +350,7 @@ def read_fixed(writers_schema, readers_schema, decoder)
       end
 
       def read_enum(writers_schema, readers_schema, decoder)
-        index_of_symbol = decoder.read_int
+        index_of_symbol = decoder.read_enum
```

Comment:
Should this pass the writer's schema so that the JSON decoder can find the appropriate offset?